src/server.rs

```diff
@@ -0,0 +1,294 @@
+use std::env;
+use fresh_eyes::{
+    extract_pr_details, get_pull_request_reviews, Branch as LibBranch,
+    ForkRequest as LibForkRequest, PullRequest as LibPullRequest,
+};
+use fresheyes::git_hub_service_server::{GitHubService, GitHubServiceServer};
+use fresheyes::{Branch, ForkRequest, ForkResult, PrResponse, PullRequest, PullRequestDetails};
+use tonic::{transport::Server, Request, Response, Status};
+use dotenv::dotenv;
+
+pub mod fresheyes {
+    tonic::include_proto!("fresheyes");
+}
+
+#[derive(Debug, Default)]
+pub struct GitHubServiceImpl {
+    github_token: String,
+}
+
+impl GitHubServiceImpl {
+    pub fn new(github_token: String) -> Self {
+        Self { github_token }
+    }
+}
+#[tonic::async_trait]
+impl GitHubService for GitHubServiceImpl {
+    // Implementation of the fork_repository method
+    async fn fork_repository(
+        &self,
+        request: Request<ForkRequest>,
+    ) -> Result<Response<ForkResult>, Status> {
+        let fork_request = request.into_inner();
+
+        let fork_request = LibForkRequest {
+            owner: &fork_request.owner.clone(),
+            repo: &fork_request.repo.clone(),
+        };
+
+        // Call the fork method to fork the repository on GitHub
+        match fork_request.fork().await {
+            Ok(data) => {
+                // If the fork was successful, create a ForkResult instance
+                let fork_result = ForkResult {
+                    owner: data.owner,
+                    repo: data.repo,
+                    forked_repo: data.forked_repo,
+                };
+                Ok(Response::new(fork_result))
+            }
+            Err(e) => {
+                // If there was an error, return it
+                Err(Status::internal(format!(
+                    "Failed to fork repository: {}",
+                    e
+                )))
+            }
+        }
+    }
+
+    async fn create_branch(&self, request: Request<Branch>) -> Result<Response<Branch>, Status> {
+        let branch = request.into_inner();
+
+        let branch = LibBranch {
+            owner: &branch.owner,
+            repo: &branch.repo,
+            branch_ref: &branch.branch_ref,
+            sha: &branch.sha,
+        };
+
+        match branch.create().await {
+            Ok(data) => {
+                let branch = Branch {
```

Comment:
I don't think we need to return create and this branch struct. We could just return the complete json data instead of partial information.